docs/07_how-to/01_auth.mdx

```diff
@@ -0,0 +1,395 @@
+
+## How does it work?
+Authorisation for data is based on Right Codes that you define for your application.  
+
+All users have one or more User Profiles. Each User Profile has one or more Right Codes.
+
+Once you have set up the Right Codes for your application, you can add the relevant Profiles through the front end:
+
+1. Create your Profiles: for example, **Trader**, **Senior Trader** or **Operations**.
+2. Assign one or more Right Codes to each Profile. When a user is given a Profile, she or he has all the rights belonging to that Profile.
+3. A user can be assigned more than one Profile.
+
+For example, you could set up the following user profiles:
+
+- Trader - with the Right Codes **InsertTrade** and **AmendTrade**
+- Senior Trader - with only one Right Code: **CancelTrade**
+
+You could then give all the traders on the team the User Profile Trader. This includes the head trader in the team, but you then also give that user the User Profile Senior Trader. This gives your head trader a superset of rights.
+
+Let's look at how these Right Codes work at different levels.
+
+### Restricting access to tables (entities)
+
+First, let us look at how you control access to specific tables.
+
+The example below restricts who can view a Data Server query called ALL_TRADES. Only users who have a User Profile containing the Right Code **TradeView** can view the data in this table. Users without this Right Code are prevented from calling this query - so the data is fully protected:
+
+```jsx
+dataServer {
+  query("ALL_TRADES", TRADE) {
+    permissioning {
+      permissionCodes = listOf("TradeView")
+    }
+  fields {
+	  ...
+	  }
+  }
+}
+```
+
+Events are the mechanism for changing the database. You can control access to these in the same way.
+
+In the example below, only users with the Right Code **TradeInsert** in their profile are able to insert trades.  Attempts to insert by all other users will be rejected.
+
+```jsx
+eventHandler<Trade>("TRADE_INSERT", transactional = true) {
+	permissioning {
+		permissionCodes = listOf("TradeInsert")
+	}
+	...
+}
+```
+
+### Restricting access to rows and columns within a table
+
+This is where you use Right Codes to control access to rows and columns.
+
+For example, you can hide a column if a user does not have a specific Right Code in their profile.
+
+In the example below, the code uses `auth` and `hideFields` to check if the user's Profile has the Right Code **TradeViewFull**. If this code is not in the Profile, then the column is not returned by the query:
+
+use :
+```jsx
+  query("ALL_TRADES", TRADE) {
+    permissioning {
+      auth {
+        hideFields { userName, rowData ->
+          if (!userHasRight(userName, "TradeViewFull")) listOf(CUSTOMER_NAME)
+          else emptyList()
+        }
+      }
+    }
+  }
+```
+
```

Comment:
We should note to say this is for dataservers in this example, but same can be applied to request servers

Reply:
its under mass authorisation but this section is same heading level and separate heading